### PR TITLE
Add setting to limit number of historical messages to load on connect

### LINKED ISSUE #2250: Add setting for limiting how many historical messages to fetch from the recent-messages api
**What should be added?**
The ability to limit how many messages are fetched for each channel from the Recent Messages API. This can be achieved by passing the ?limit query parameter to the recent messages API.

**Why should it be added?**
Performance - in case your computer doesn't like you fetching up-to 800 messages (the default) from the API.

API Docs: https://recent-messages.robotty.de/api

### PR DESCRIPTION
**Pull request by pajlada**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

This pull request adds a setting to limit the amount of historical messages to fetch from the Recent Messages API.

Fixes #2250

### CODE REVIEW

**Comment by RAnders00** [comment]
Downloaded the windows build, works fine for me 👍

File: src/widgets/settingspages/GeneralPage.cpp [hunk 1]
```diff
@@ -600,6 +600,9 @@ void GeneralPage::initLayout(GeneralPageView &layout)
                        s.highlightInlineWhispers);
     layout.addCheckbox("Load message history on connect",
                        s.loadTwitchMessageHistoryOnConnect);
+    // TODO: Change phrasing to use better english once we can tag settings, right now it's kept as history instead of historical so that the setting shows up when the user searches for history
+    layout.addIntInput("Max number of history messages to load on connect",
+                       s.twitchMessageHistoryLimit, 10, 800, 10);
```
Comment (revolter) [inline src/widgets/settingspages/GeneralPage.cpp:605]: @pajlada, Why is the maximum limit only `800`?

Comment (zneix) [reply src/widgets/settingspages/GeneralPage.cpp:605]: IIRC this is how much randers' service returns at max, so there's no need for a higher limit.
Also, loading ~950 or more messages crashes Chatterino, see #2226

Comment (killertofus) [reply src/widgets/settingspages/GeneralPage.cpp:605]: > 
> 
> IIRC this is how much randers' service returns at max, so there's no need for a higher limit.
> Also, loading ~950 or more messages crashes Chatterino, see #2226

you meant #1421

**Pull request merged**